test/index.spec.js

```diff
@@ -225,3 +225,73 @@ it('should handle the provided query and query options objects', () => {
 
     expect(urlJoin('/google.com', options)).toBe('/google.com?foo[0]=1&foo[1]=2&foo[2]=3');
 });
+
+it('should keep the trailing slash', () => {
+    const options = { trailingSlash: 'keep' };
+
+    expect(urlJoin(options)).toBe('');
```

Comment:
This is still failing though @satazor

Reply:
Sorry, that test is wrong as well. Because, `leadingSlash` is true by default, it should be `/`.

See test on line 4.

Reply:
@guilhermedeandrade I've updated the gist, fixing a few bugs in the code and also added a test file based on your tests but with more combinations.

Reply:
Ok, fixed!

Reply:
@satazor Oh sorry, when I was typing my last comment your latest one had not appeared yet. I will update and push the changes

Reply:
Maybe I didn't updated at all. I've pushed the gist now.

Reply:
Just pushed your changes @satazor